### ISSUE TITLE
feat: dialect Node.js

### ISSUE THREAD

**Issue by jlarmstrongiv** [subframe7536/kysely-sqlite-tools#13]
https://nodejs.org/api/sqlite.html

**Comment by jlarmstrongiv** [comment]
There are two types of classes:
- DatabaseSync 
- DatabaseAsync

DatabaseSync is currently available and is similar to better-sqlite3 and bun adapters

**Comment by subframe7536** [comment]
Yeah, that looks promising, and I found it months ago. But I decide not to implement it until the apis are stable and nobody have already implement the dialect.

**Comment by mmgoodnow** [comment]
haven't played with any migrations yet, but I took a stab here. It relies on VERY new versions of node that aren't yet typed. https://github.com/mmgoodnow/kysely-node-sqlite/blob/df086d566e9e9cefb0b6d8edc253a8cd428c8d2b/dialect/sqlite-dialect.ts

The differences are pretty minimal - 

- `StatementSync.reader` doesn't seem to exist - I replaced with `SelectQueryNode.is` which I'm not sure is right generally, but seems right enough for my purposes 🤷 
- you have to spread the params array into arguments on StatementSync methods

**Comment by subframe7536** [comment]
@mmgoodnow One thing that can be optimized: even if `SelectQueryNode.is(query)` returns `false`, you still need to use `stmt.all()` instead of `stmt.run()` to get the result from `db.insertInto('table').values({data: 'test'}).returning('data')`

**Comment by subframe7536** [comment]
The generic sqlite dialect is released and you can create it follow to guide or see the usage of other dialects in this repo to make the dialect.

And sorry, but currently I have no extra bandwidth working on it.

**Comment by subframe7536** [comment]
Here is the implemention: https://github.com/kysely-org/kysely/issues/1292#issuecomment-2670341588
